docs/site/tutorials/todo-list/todo-list-tutorial-sqldb.md

```diff
@@ -24,22 +31,689 @@ rm src/datasources/db.datasource.*
   REST services (supported by StrongLoop)
   ...
 ```
 
+Your db.datasource.ts file should look like this:
+
+{% include code-caption.html content="/src/datasources/db.datasource.ts" %}
+
+```json
+const config = {
+  name: 'db',
+  connector: 'mysql',
+  url: '',
+  host: ',
+  port: 0,
+  user: '',
+  password: '',
+  database: '',
+};
+```
+
+### Specifying the datasource properties
+
+Please fill in the `host`, `port`, `user`, and `password` based on whether:
+
+- you already have your own MySQL installation
+- you followed the steps in
+  [Setting up MySQL in Docker](./todo-list-tutorial-docker-mysql.md),
+
+For the purposes of this tutorial, please set the `database` property to
+`testdb`.
+
+The MySQL connector has an extra property, `createDatabase: true`, we can add to
+the datasource settings to create the database for us if it doesn't already
+exist. (If you are using a different relational database, you will need to
+manually create the database before attempting to connect to it.)
+
+The final settings for the datasource for connecting to the Docker MySQL setup:
+
+{% include code-caption.html content="/src/datasources/db.datasource.ts" %}
+
+```json
+const config = {
+  name: 'db',
+  connector: 'mysql',
+  url: '',
+  host: '127.0.0.1',
+  port: 3306,
+  user: 'root',
+  password: 'my-secret-pw',
+  database: 'testdb',
+  createDatabase: true,
+};
+```
+
+See [MySQL connector](https://loopback.io/doc/en/lb4/MySQL-connector.html) for
+detailed information about different settings you can specify here, but what we
+currently have is sufficient.
+
+### Change id property's 'generated' setting to true for all models
+
+Currently the `Todo`, `TodoList`, and `TodoListImage` models have an `id`
+property with a setting of `generated: false` .
+
+```json
+ @property({
+    type: 'number',
+    id: true,
+    generated: false,
+  })
+  id?: number;
+```
+
+We need to change the `id` property's `generated` setting to `true` for all
+these models in order for the create and update CRUD functions to work in a
+MySQL database.
+
+```json
+ @property({
+    type: 'number',
+    id: true,
+    generated: true,
+  })
+  id?: number;
+```
+
+### Creating the database tables
+
+Every LoopBack application contains a `/src/migrate.ts` file.
+
+```ts
+import {TodoListApplication} from './application';
+
+export async function migrate(args: string[]) {
+  const existingSchema = args.includes('--rebuild') ? 'drop' : 'alter';
+  console.log('Migrating schemas (%s existing schema)', existingSchema);
+
+  const app = new TodoListApplication();
+  await app.boot();
+  await app.migrateSchema({existingSchema});
+
+  // Connectors usually keep a pool of opened connections,
+  // this keeps the process running even after all work is done.
+  // We need to exit explicitly.
+  process.exit(0);
+}
+
+migrate(process.argv).catch(err => {
+  console.error('Cannot migrate database schema', err);
+  process.exit(1);
+});
+```
+
```

Comment:
I think it might be good to briefly mention how to [auto-migrate/auto-update models with foreign keys](https://loopback.io/doc/en/lb4/MySQL-connector.html#auto-migrateauto-update-models-with-foreign-keys) here.

Reply:
> auto-migrate/auto-update models with foreign keys

Hey Agnes. Thx. I will put it in the `Specify the foreign key constraint in the models` section lower in the document.